testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -144,7 +147,7 @@ public void testParseError()
     public void testPlanningFailure()
             throws Exception
     {
-        assertFailedQuery("SELECT * FROM mock.default.tests_table", "Throw from apply projection");
+        assertFailedQuery("SELECT test + 1 FROM mock.default.tests_table", "Throw from apply projection");
```

Comment:
Why it does not fail with `*`?

Reply:
With `*` PushProjectionIntoTableScan doesn't run, it seems the projection gets pruned before that rule gets a chance to run.

Reply:
would `test` itself be sufficient? Why we need `+1`?

Reply:
even with just `test`, the PushProjectionIntoTableScan doesn't trigger, putting a non-trivial expression in the projection was the only way I was able to get the ProjectNode to stay until PushProjectionIntoTableScan rule is triggered.